lib/cc/analyzer/container.rb

```diff
@@ -104,17 +102,22 @@ def read_stderr(err)
         end
       end
 
-      def timeout_thread(pid)
+      def timeout_thread
         Thread.new do
           sleep @timeout
           @timed_out = true
-          Process.kill("KILL", pid)
+          reap_running_container
         end
       end
 
       def container_data(duration: nil, status: nil)
         ContainerData.new(@image, @name, duration, status, @stderr_io.string)
       end
+
+      def reap_running_container
+        Analyzer.logger.warn("killing container name=#{@name}")
+        POSIX::Spawn::Child.new("docker", "kill", @name)
```

Comment:
I would actually be against it raising if it fails because we're already in an error-handler of sorts and you'd end up masking what really happened with that failure -- maybe that'd be OK, I don't know.